### PR TITLE
fix: small random openapi parser fixes

### PR DESCRIPTION
**Pull request by armandobelardo**
the first fixes inlining objects that are only used in one request
the second allows generic objects to be used in query parameters

### CODE REVIEW

File: packages/cli/openapi-ir-to-fern/src/buildEndpoint.ts [hunk 1]
```diff
@@ -273,17 +273,33 @@ function getRequest({
...
+                    fileContainingReference: declarationFile,
+                    context
+                });
+                convertedRequest = {
+                    schemaIdsToExclude: [],
+                    value: {
+                        body: requestTypeReference
+                    }
+                };
+            }
```
Comment (dsinghvi) [inline packages/cli/openapi-ir-to-fern/src/buildEndpoint.ts:302]: @armandobelardo how does this end up inlining the schema?

**Review by dsinghvi — APPROVED** [review]
approving to unblock you but any context on why the `buildEndpoint.ts` change works would be great

File: packages/cli/openapi-ir-to-fern/src/buildEndpoint.ts [hunk 2]
```diff
@@ -273,17 +273,33 @@ function getRequest({
...
-            const convertedRequest: ConvertedRequest = {
-                schemaIdsToExclude: [],
-                value: {
-                    body: requestTypeReference
-                }
-            };
+            let requestTypeReference;
+            let convertedRequest: ConvertedRequest;
+            if (maybeSchemaId === generatedRequestName && !nonRequestReferencedSchemas.includes(maybeSchemaId)) {
+                requestTypeReference = buildTypeReference({
```
Comment (dsinghvi) [inline packages/cli/openapi-ir-to-fern/src/buildEndpoint.ts:279]: `buildTypeDeclaration`

**Comment by dannysheridan** [comment]
@armandobelardo stale PR, merge or close?

**Pull request closed without merging**